channeldb/channel.go

```diff
@@ -224,8 +224,17 @@ type OpenChannel struct {
 	// aren't yet able to verify that it's actually in the hash chain.
 	TheirCurrentRevocation     *btcec.PublicKey
 	TheirCurrentRevocationHash [32]byte
-	LocalElkrem                *elkrem.ElkremSender
-	RemoteElkrem               *elkrem.ElkremReceiver
+
+	// PreimageProducer is used to generate the revocation in such a way
+	// that remote side might store it efficiently and have the ability to
+	// restore the revocation by index if needed. Current implementation of
+	// secret producer is shachain producer.
+	PreimageProducer shachain.SecretProducer
```

Comment:
I agree, it would be better to rename `PreimageProducer` to `RevocationProducer`. But I would rather not rename the `SecretProducer` to `RevocationReceiver`, the ideal case might be rename both the variable and interface to `PreimageProducer`. 

The words `Producer/Store` reflect the true essence/goal of `elkrem`, `shachain` algorithms - produce the revocation key in such a way that it might be stored and recovered on other side in efficient manner. The prefixes `Sender`, `Receiver` not reflects this meaning `IMHO`, and as to newcomer it was a bit confusing for me to understand why `sender` and `receiver` generating revocation hashes and store them.